source/MaterialXContrib/MaterialXNode/MaterialXNode.cpp

```diff
@@ -138,6 +139,19 @@ bool MaterialXNode::setInternalValue(const MPlug& plug, const MDataHandle& dataH
         );
     };
 
+    auto createAndRegister = [this](mx::DocumentPtr document)
+    {
+        _materialXData.reset(new MaterialXData(
+            document,
+            _elementPath.asChar(),
+            Plugin::instance().getLibrarySearchPath()
+        ));
+
+        MaterialXMaya::registerFragment(
```

Comment:
Registration when either the document or the element changes. Still have to generate unique fragment names.